terraform/Makefile

```diff
@@ -30,10 +30,16 @@ plan: validate
 
 .PHONY: apply
 apply: validate rename_resources
+	$(if $(findstring 1,$(AZUL_PRIVATE_API)), \
+	terraform apply -target aws_vpc_endpoint.indexer -target aws_vpc_endpoint.service \
+	)
 	terraform apply
 
 .PHONY: auto_apply
 auto_apply: validate rename_resources
+	$(if $(findstring 1,$(AZUL_PRIVATE_API)),\
+		terraform apply -auto-approve -target aws_vpc_endpoint.indexer -target aws_vpc_endpoint.service \
+	)
```

Comment:
These resources need to be created first so that the `for_each` in the `aws_network_interface` data source (api_gateway.tf.json.template.py) will be able to iterate the network interface ids of the `aws_vpc_endpoint`s.

Reply:
[This](https://github.com/DataBiosphere/azul/pull/4144/commits/1a711d111053ca4a3073e1f2ced2647201ed338b#r894768964) should be explained in a comment here and in the file section referenced by your answer to the reviewers question.